### PR TITLE
new contents of 1.1-vector

### PR DESCRIPTION
**Pull request by lacrimosaprinz**
## What type of PR is this?

- [ ] Enhancement
- [ ] Displaying
- [ ] Typo
- [x] Doc Request

## Which issue(s) this PR fixes:

issue #
https://github.com/matrixorigin/matrixone/issues/11563

### CODE REVIEW

File: docs/MatrixOne/Reference/Functions-and-Operators/1.1-Vector/misc.md [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+drop table if exists vec_table;
+create table vec_table(a int, b vecf32(3), c vecf64(3));
+insert into vec_table values(1, "[1,2,3]", "[4,5,6]");
+mysql> select * from vec_table;
++------+-----------+-----------+
+| a    | b         | c         |
++------+-----------+-----------+
+|    1 | [1, 2, 3] | [4, 5, 6] |
++------+-----------+-----------+
+1 row in set (0.00 sec)
```
Comment (arjunsk) [inline docs/MatrixOne/Reference/Functions-and-Operators/1.1-Vector/misc.md:76]:
```mysql
insert into vec_table values(1, "[-1,-2,3]", "[4,5,6]");

mysql> select * from vec_table;
+------+-----------+-----------+
| a    | b         | c         |
+------+-----------+-----------+
|    1 | [-1, -2, 3] | [4, 5, 6] |
+------+-----------+-----------+

```

File: docs/MatrixOne/Reference/Functions-and-Operators/1.1-Vector/misc.md [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+
+mysql> select abs(cast("[-1,-2,3]" as vecf32(3)));
```
Comment (arjunsk) [inline docs/MatrixOne/Reference/Functions-and-Operators/1.1-Vector/misc.md:122]: Let's use this example

```mysql
mysql> select b + cast("[1,2,3]" as vecf32(3)) from vec_table;
+--------------------------------+
| b + cast([1,2,3] as vecf32(3)) |
+--------------------------------+
| [2, 4, 6]                      |
+--------------------------------+
```

File: docs/MatrixOne/Reference/Functions-and-Operators/1.1-Vector/vector_dims.md [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
++------+-----------+-----------+
+1 row in set (0.00 sec)
+
+mysql> select vector_dims(b) from vec_table;
++----------------+
+| vector_dims(b) |
++----------------+
+|              3 |
++----------------+
+1 row in set (0.01 sec)
```
Comment (arjunsk) [inline docs/MatrixOne/Reference/Functions-and-Operators/1.1-Vector/vector_dims.md:33]: Let's use this example.

```mysql


insert into vec_table values(1, "[1,2,3]", "[4,5,6]");
insert into vec_table values(2, "[7,8,9]", "[1,2,3]");
mysql> select * from vec_table;
+------+-----------+-----------+
| a    | b         | c         |
+------+-----------+-----------+
|    1 | [1, 2, 3] | [4, 5, 6] |
|    2 | [7, 8, 9] | [1, 2, 3] |
+------+-----------+-----------+
2 row in set (0.00 sec)

mysql> select vector_dims(b) from vec_table;
+----------------+
| vector_dims(b) |
+----------------+
|              3 |
|              3 |
+----------------+
2 row in set (0.01 sec)
```

File: docs/MatrixOne/Reference/Functions-and-Operators/1.1-Vector/vector_dims.md [hunk 4]
```diff
@@ -0,0 +1,34 @@
+# **vector_dims()**
+
+## **Description**
```
Comment (arjunsk) [inline docs/MatrixOne/Reference/Functions-and-Operators/1.1-Vector/vector_dims.md:3]: Missing Best Practises.

File: docs/MatrixOne/Develop/schema-design/1.1-vector.md [hunk 5]
```diff
@@ -0,0 +1,143 @@
+# Vectors
+
+## What Are Vectors
+
+In a database, vectors are typically sets of numbers arranged in a specific way to represent some form of data or features. These vectors can be one-dimensional, multidimensional, or data structures with higher dimensions. In the fields of machine learning and data analysis, vectors are used to represent data points, features, or model parameters.
```
Comment (arjunsk) [inline docs/MatrixOne/Develop/schema-design/1.1-vector.md:5]: needs change

**Review by arjunsk — CHANGES_REQUESTED** [review]
Please correct the review comments

File: docs/MatrixOne/Develop/schema-design/1.1-vector.md [hunk 6]
```diff
@@ -0,0 +1,143 @@
...
+
+## Advantages of Vectors
+
+The capability of a database to work with vectors means that the database system possesses the ability to store, query, and analyze vector data. These vectors are often associated with complex data analysis, machine learning, and mining tasks. Here are some advantages of a database having vector capabilities:
```
[the remark this answers is not in the document]
Comment (lacrimosaprinz) [reply docs/MatrixOne/Develop/schema-design/1.1-vector.md:9]: reedit the content: A database's capacity to handle vectors signifies its ability to store, query, and analyze vector data. These vectors frequently play a crucial role in intricate data analysis, machine learning, and mining endeavors. Here are several benefits associated with a database that possesses vector capabilities:

File: docs/MatrixOne/Develop/schema-design/1.1-vector.md [hunk 7]
```diff
@@ -0,0 +1,143 @@
...
+SELECT * FROM t1 ORDER BY l2_norm(b - '[3,1,2]') LIMIT 5;
+
+-- Top K Queries using cosine similarity
+SELECT * FROM t1 ORDER BY cosine_similarity(b, '[3,1,2]') LIMIT 5;
+
+-- Top K Queries using cosine distance
+SELECT * FROM t1 ORDER BY 1 - cosine_similarity(b, '[3,1,2]') LIMIT 5;
+```
+
+These queries demonstrate retrieving the top 5 vectors most similar to the given vector `[3,1,2]` using different distance and similarity measures. Here, `l1_distance` and `l2_distance` represent L1 distance and L2 distance, respectively. `cosine_similarity` represents cosine similarity, and `cosine_distance` represents cosine distance. With these queries, you can find the data that best matches your target vector based on different measurement criteria.
```
[the remark this answers is not in the document]
Comment (lacrimosaprinz) [reply docs/MatrixOne/Develop/schema-design/1.1-vector.md:125]: I deleted

File: docs/MatrixOne/Develop/schema-design/1.1-vector.md (hunk 5 above)
Comment (lacrimosaprinz) [reply docs/MatrixOne/Develop/schema-design/1.1-vector.md]: I reedited: 
In a database, vectors typically consist of organized sets of numbers arranged in a specific manner to represent various data or characteristics. These vectors can exist in one-dimensional, multidimensional, or even higher-dimensional data structures. In machine learning and data analysis, vectors represent data points, features, or model parameters.

File: docs/MatrixOne/Develop/schema-design/1.1-vector.md [hunk 8]
```diff
@@ -0,0 +1,176 @@
+# Vectors
+
+## What Are Vectors
+
```
Comment (arjunsk) [inline docs/MatrixOne/Develop/schema-design/1.1-vector.md:4]: Data content, including text and audio, can be transformed into numerical representations known as vector embeddings using embedding models. These embeddings can be stored in databases and queried to find nearest neighbors, enhancing content search capabilities. Additionally, language models can augment this functionality by generating data based on these embeddings.

File: docs/MatrixOne/Develop/schema-design/1.1-vector.md [hunk 9]
```diff
@@ -0,0 +1,176 @@
...
+In a database, vectors typically consist of organized sets of numbers arranged in a specific manner to represent various data or characteristics. These vectors can exist in one-dimensional, multidimensional, or even higher-dimensional data structures. In machine learning and data analysis, vectors represent data points, features, or model parameters.
+
+## Advantages of Vectors
+
+A database's capacity to handle vectors signifies its ability to store, query, and analyze vector data. These vectors frequently play a crucial role in intricate data analysis, machine learning, and mining endeavors. Here are several benefits associated with a database that possesses vector capabilities:
```
Comment (arjunsk) [inline docs/MatrixOne/Develop/schema-design/1.1-vector.md:9]: We can skip the Advantages of Vectors section.

File: docs/MatrixOne/Develop/schema-design/1.1-overview.md [hunk 10]
```diff
@@ -0,0 +1,80 @@
...
+The indexes commonly used in databases include primary key indexes, secondary indexes, etc. Among them, unique indexes are used to ensure the uniqueness of specific columns or combinations of columns, ordinary indexes are used to improve query performance, and full-text indexes are used for full-text search in text data.
+
+There are two common types of indexes, namely:
+
+- **Primary Key**: Primary key index, that is, the index that identifies the primary key column. The primary key index uniquely identifies each row of data in the table.
+- **Secondary index**: Secondary index is the index identified on the non-primary key. The secondary index, also called a non-clustered index, is used to improve query performance and speed up data retrieval.
+
+## Vector
+
+MatrixOne now supports storing and querying vectors. Vector is a list of numbers typically generated by AI models, such as Large Language Models. The correlation between two vectors is measured by measuring their distance, with smaller distances indicating higher similarity.
```
Comment (arjunsk) [inline docs/MatrixOne/Develop/schema-design/1.1-overview.md:36]: MatrixOne now supports storing and querying vectors. Vector is a numerical array generally produced by AI models, including Large Language Models.  These vectors can be seamlessly stored and queried, allowing for tasks like finding nearest neighbors, all while accommodating relational data.

File: docs/MatrixOne/Develop/schema-design/1.1-vector.md [hunk 11]
```diff
@@ -0,0 +1,176 @@
...
++--------------------------+
+| 0000803f0000004000004040 |
+| 7e98b23e9e10383b2f41133f |
++--------------------------+
+2 rows in set (0.00 sec)
+```
+
+## Example - Top K Queries
+
+Top K queries are a database query operation that retrieves the top K data items or records from a database. Top K queries can be applied in various scenarios, including recommendation systems, search engines, data analysis and sorting.
```
Comment (arjunsk) [inline docs/MatrixOne/Develop/schema-design/1.1-vector.md:98]: Top K queries are a database query operation that retrieves the top K data items or records from a database. These queries find application in a wide variety of scenarios, including recommendation systems, search engines, and data analysis, among other fields.

File: docs/MatrixOne/Develop/schema-design/1.1-overview.md [hunk 12]
```diff
@@ -0,0 +1,80 @@
...
+## Index
+
+An index is a data structure used to find data in database tables quickly. It can be seen as a 'table of contents' that contains pointers to the data of each row in the table, making it possible for queries to locate data that meets specific conditions more quickly.
+
+The indexes commonly used in databases include primary key indexes, secondary indexes, etc. Among them, unique indexes are used to ensure the uniqueness of specific columns or combinations of columns, ordinary indexes are used to improve query performance, and full-text indexes are used for full-text search in text data.
+
+There are two common types of indexes, namely:
+
+- **Primary Key**: Primary key index, that is, the index that identifies the primary key column. The primary key index uniquely identifies each row of data in the table.
+- **Secondary index**: Secondary index is the index identified on the non-primary key. The secondary index, also called a non-clustered index, is used to improve query performance and speed up data retrieval.
```
[the remark this answers is not in the document]
Comment (lacrimosaprinz) [reply docs/MatrixOne/Develop/schema-design/1.1-overview.md]: I reedited: MatrixOne supports one type of index, namely:

- **Primary Key**: Primary key index, the index that identifies the primary key column. The primary key index uniquely identifies each row of data in the table.

File: docs/MatrixOne/Develop/schema-design/1.1-vector.md (hunk 8 above)
Comment (lacrimosaprinz) [reply docs/MatrixOne/Develop/schema-design/1.1-vector.md:4]: I tweaked a few words: A vector is a numerical representation encodes data content, such as text or audio, using embedding models. These vectors can be effectively stored and queried alongside relational data. Vector queries encompass tasks like locating nearest neighbors, which can significantly improve search retrievals, as seen in applications like facial recognition or enhancing Generative AI results.

File: docs/MatrixOne/Develop/schema-design/1.1-vector.md (hunk 9 above)
Comment (lacrimosaprinz) [reply docs/MatrixOne/Develop/schema-design/1.1-vector.md:9]: I tweaked a few words: 
```
## Vector database Use Cases

The database's capability to manage vectors reflects its capacity to store, retrieve, and analyze vector-based data. These vectors often play a pivotal role in complex data analysis, machine learning, and mining projects. Vector databases offer a wide range of versatile use cases:

- **Generative AI Applications**: These databases can serve as a backend for Generative AI applications, enabling them to fetch the nearest neighbor results in response to user-provided queries, enhancing the output quality and relevance.

- **Advanced Object Recognition**: They are invaluable for developing advanced object recognition platforms that discern similarities among diverse datasets. This has practical applications in plagiarism detection, facial recognition, and DNA matching.

- **Personalized Recommendation Systems**: Vector databases can be leveraged to augment recommendation systems by incorporating user preferences and choices. This leads to more accurate and tailored recommendations, improving the user experience and engagement.

- **Anomaly Detection**: Vector databases can store feature vectors representing normal behavior. Anomalies can then be detected by comparing incoming vectors to the stored ones. This is useful in cybersecurity and industrial quality control.
```

**Review by arjunsk — APPROVED** [review]

**Review by dengn — APPROVED** [review]

File: docs/MatrixOne/Reference/Functions-and-Operators/1.1-Vector/cosine_similarity.md [hunk 13]
```diff
@@ -0,0 +1,41 @@
...
++-------------------------------+
+|                             1 |
++-------------------------------+
+1 row in set (0.00 sec)
+```
+
+## **Constraints**
+
+- Both the argument vectors should have same dimensions
+- Cosine similarity value lies between -1 and 1.
```
Comment (arjunsk) [inline docs/MatrixOne/Reference/Functions-and-Operators/1.1-Vector/cosine_similarity.md:41]: - Cosine similarity value lies between 0 and 1.

**Pull request merged**